src/ops/noop.cc

```diff
@@ -166,23 +162,25 @@ Node FFModel::get_or_create_noop_node(const ParallelTensor input)
     cached_noop_ops[hash] = noop;
   }
   Node ret;
-  ret.guid = node_global_guid ++;
+  ret.guid = node_global_guid++;
   ret.ptr = noop;
   return ret;
 }
 
-Node FFModel::get_or_create_input_node(const ParallelTensorShape& output_shape)
-{
-  size_t hash = std::hash<ParallelTensorShape>{}(output_shape); 
-  NoOp* input = NULL;
-  const auto& it = cached_input_ops.find(hash);
+Node FFModel::get_or_create_input_node(
```

Comment:
This should ideally just forward to the generic `FFModel::get_or_create_node<T>` method (see `Linear` for an example)